source/common/ssl/context_impl.cc

```diff
@@ -223,6 +223,23 @@ bool ContextImpl::verifySubjectAltName(X509* cert,
   return verified;
 }
 
+bool ContextImpl::uriMatch(const std::string& uriPattern, const char* uri) {
```

Comment:
got it. Since my original `strlen()` was unnecessary, I'll change back to `string::compare()` to match c++11 style so that future change will be easier.